homeassistant/components/matter/number.py

```diff
@@ -365,6 +365,25 @@ def _update_from_device(self) -> None:
             clusters.MicrowaveOvenControl.Attributes.MaxCookTime,
         ),
     ),
+    MatterDiscoverySchema(
+        platform=Platform.NUMBER,
+        entity_description=MatterNumberEntityDescription(
+            key="speaker_setpoint",
+            translation_key="speaker_setpoint",
+            native_unit_of_measurement=PERCENTAGE,
+            native_max_value=100,
+            native_min_value=0,
+            native_step=1,
+            device_to_ha=lambda x: None
+            if x is None
+            else round(min(x, 254) * 100 / 254),  # Linear mapping: 0→0%, 254→100%
```

Comment:
formally we should read min/maxLevel and use them to scale because especially for volumes it might be a 0..100 or such